### ISSUE TITLE
Adding two or more Xamarin.Firebase.xxx causes an exception because the instance cannot be obtained with Null.

### ISSUE THREAD

**Issue by MyAlfisti** [xamarin/GooglePlayServicesComponents#268]
### Xamarin.Android Version (eg: 6.0):

Oreo 8.1 (API27) - Q 10.0 (API29)

### Operating System & Version (eg: Mac OSX 10.11):

macOS 10.15

### Google Play Services Version (eg: 8.4.0):

latest

### Describe your Issue:

If you add two or more Xamarin.Firebase. ~ Packages, the instance cannot be obtained with Null and an exception occurs.

### Steps to Reproduce (with link to sample solution if possible):

1. Added Xamarin.Firebase.Auth.
2. Implement authentication.
3. Processing is completed normally.
4. Added Xamarin.Firebase.Firestore.
5. Initialize FirebaseApp.
var baseOptions = FirebaseOptions.FromResource (context);
var options = new FirebaseOptions.Builder (baseOptions) .SetProjectId (baseOptions.StorageBucket.Split ('.') [0]). Build ();
_app ?? = FirebaseApp.InitializeApp (context, options, baseOptions.ProjectId);

6. Initialize FirebaseFirestore. An exception occurs here.
FirebaseFirestore.GetInstance (_app);

7. Removed Xamarin.Firebase.Auth.
8.
FirebaseFirestore.GetInstance (_app) in steps 5-6
Succeeds.

### Include any relevant Exception Stack traces, build logs, adb logs:

Java.Lang.NullPointerException: Firestore component is not present.
  at Java.Interop.JniEnvironment+StaticMethods.CallStaticObjectMethod (Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args) [0x00069] in <42bcf67b56bc4c909c2a5edee682522b>:0
  at Java.Interop.JniPeerMembers+JniStaticMethods.InvokeObjectMethod (System.String encodedMember, Java.Interop.JniArgumentValue* parameters) [0x00018] in <42bcf67b56bc4c909c2a5edee682522b>:0
  at Firebase.Firestore.FirebaseFirestore.GetInstance (Firebase.FirebaseApp app) [0x00031] in <67ce9d11507544239e590d0416a617d6>:0
  at Scorecard.Droid.Services.FirestoreService.Initialize (Android.Content.Context context) [0x0004f] in /Users/g0ld3/Documents/source/Scorecard/Scorecard.Android/Services/FirestoreService.cs:20
  at Scorecard.Droid.MainActivity.OnCreate (Android.OS.Bundle savedInstanceState) [0x00074] in /Users/g0ld3/Documents/source/Scorecard/Scorecard.Android/MainActivity.cs:29
  at Android.App.Activity.n_OnCreate_Landroid_os_Bundle_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_savedInstanceState) [0x00011] in <07ae79e8d8f94e5fa94a23fe57b46db3>:0
  at at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.8(intptr,intptr,intptr)
  at java.lang.NullPointerException: Firestore component is not present.
  at at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:910)
  at at com.google.firebase.firestore.FirebaseFirestore.zza(com.google.firebase:firebase-firestore@@17.0.5:66)
  at at com.google.firebase.firestore.FirebaseFirestore.getInstance(com.google.firebase:firebase-firestore@@17.0.5:58)
  at at md51193cfde975901f47a3f76b392c90a44.MainActivity.n_onCreate(Native Method)
  at at md51193cfde975901f47a3f76b392c90a44.MainActivity.onCreate(MainActivity.java:30)
  at at android.app.Activity.performCreate(Activity.java:7802)
  at at android.app.Activity.performCreate(Activity.java:7791)
  at at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1299)
  at at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
  at at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
  at at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
  at at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
  at at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
  at at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
  at at android.os.Handler.dispatchMessage(Handler.java:107)
  at at android.os.Looper.loop(Looper.java:214)
  at at android.app.ActivityThread.main(ActivityThread.java:7356)
  at at java.lang.reflect.Method.invoke(Native Method)
  at at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
  at at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)

**Comment by moljac** [comment]
@MyAlfisti 

Could you try `Xamarin.Android.ManifestMerger` and provide us feedback, please?
There are known issues with `AndroidManifest.xml` and we are testing new approach. 

https://www.nuget.org/packages/Xamarin.Android.ManifestMerger/1.0.0-preview03

**Comment by moljac** [comment]
@MyAlfisti Can you provide feedback info whether ManifestMerger nuget helped?
Thanks

**Comment by JPZV** [comment]
@moljac I was having the same problem (my stack was the same but shorter, but the message was the same). I can confirm that ManifestMerger fixed the problem.

I have to note that my project has two parts, an assembly project (that I call as _Core_) and the App itself (_App_) where the _App_ references the _Core_, but also the _Core_ has everything about Firebase (_App_ doesn't has anything but Guava). In order to solve this problem, ManifestMerger must be added to the _App_ project even if it doesn't has any Firebase package directly. It's not necessary to add ManifestMerger to the _Core_ project (also adding this package only to _Core_ will not fix the problem).

In summary: ManifestMerger must be added to the final application project instead of any assembly project in the same solution.

**Comment by moljac** [comment]
@JPZV Thanks a lot for your feedback

**Comment by moljac** [comment]
@MyAlfisti 

This issue will be closed due in few days to lack of information/feedback.

**Comment by MyAlfisti** [comment]
Sorry for the late reply.
Now I give up Firebase and Firestore and implement it differently.
Therefore, it seems that feedback will be delayed because the specified method cannot be tried immediately.
Administrators should remove this issue if it is not appropriate.
